crates/net/nat/src/lib.rs

```diff
@@ -191,20 +192,26 @@ pub async fn external_addr_with(resolver: NatResolver) -> Option<IpAddr> {
     match resolver {
         NatResolver::Any | NatResolver::Upnp | NatResolver::PublicIp => resolve_external_ip().await,
         NatResolver::ExternalIp(ip) => Some(ip),
-        NatResolver::NetIf => match resolve_net_if_ip(DEFAULT_NET_IF_NAME) {
+        NatResolver::NetIf => match resolve_net_if_ip(DEFAULT_NET_IF_NAME).inspect_err(|err| {
+            debug!(target: "net::nat", "Failed to resolve network interface IP: {}", err);
```

Comment:
```suggestion
            debug!(target: "net::nat",
                 %err,
                "Failed to resolve network interface IP"
            );
```

this way we get the nice key value formatting by the tracing library **err**=_Some error message_

Reply:
Interesting thanks!